### ISSUE TITLE
dataset_benchmark should not expect targets

### ISSUE THREAD

**Issue by Srijan2-0-2-3** [ContinualAI/avalanche#1480]
The code has following lines :+1: data = dataset_benchmark(AvalancheDataset(train_dataset),AvalancheDataset(test_dataset))
It returns ValueError: Unsupported dataset: must have a valid targets field or has to be a Tensor Dataset with at least 2 Tensors

**Comment by AntonioCarta** [comment]
Thanks, this is in fact a bug. `dataset_benchmark` should work without targets. Until we fix this, you can avoid the error by adding a `targets` attribute to your dataset with class targets. If you don't have them, just add a list of zeros.

**Comment by Srijan2-0-2-3** [comment]
Thanks for the update.

**Comment by Srijan2-0-2-3** [comment]
Can you give an example for better understanding

**Comment by AntonioCarta** [comment]
something like:
```
data = my_dataset
data.targets = [0 for _ in range(len(data))]
avl_data = AvalancheDataset(data)
```
and then you can call dataset benchmark.

**Comment by Srijan2-0-2-3** [comment]
Thanks

**Comment by Srijan2-0-2-3** [comment]
So even after updating `targets` for train_data and test_data and passing in `dataset_benchmark(AvalancheDataset(train_data),Avalanche(test_data))`. It is still showing the same error as before.

**Comment by Srijan2-0-2-3** [comment]
Hello sir
I am trying to add a new dataset for continual learning. In this I am creating the above mentioned dataset benchmark. The dataset consist of data and target values. Since, it is returning the above mentioned error kindly help me in solving it.

**Comment by AntonioCarta** [comment]
I was wrong in my previous answer, you have to use [make_classification_dataset](https://avalanche-api.continualai.org/en/v0.4.0/generated/avalanche.benchmarks.utils.make_classification_dataset.html#avalanche.benchmarks.utils.make_classification_dataset)

**Comment by AntonioCarta** [comment]
fixed in 0.5.0
